EFCodeFirst/EFCodeFirst/Program.cs

```diff
@@ -0,0 +1,261 @@
+
+            try
+            {
+                var order = new Order
+                {
+                    orderDetails = orderDetails,
+                    orderStatus = true,
+                    orderDate = DateTime.Now,
+                    customer = customer
+                };
+
+                var orderList = new OrderList
+                {
+                    product = product,
+                    productQuantity = productQuantity,
+                    order = order
+                };
+
+                context.Orders.Add(order);
+                context.OrderLists.Add(orderList);
+
+                context.SaveChanges();
+
+                transaction.Commit();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Operation failes with error: {ex.Message}");
+                transaction.Rollback();
+            }
+        }
+
+        static void clearDB(Context context)
+        {
+            context.Database.ExecuteSqlCommand("DELETE FROM dbo.Suppliers");
+            context.Database.ExecuteSqlCommand("DELETE FROM dbo.Customers");
+        }
+        static void prepareDB(Context context)
+        {
+            clearDB(context);
+
+            prepareCustomers(context);
+            prepareSuppliers(context);
+            prepareProducts(context);
+            prepareOrders(context);
+        }
+        static void prepareCustomers(Context context)
+        {
+            var customer1 = new Customer
+            {
+                cutomerName = "Petya",
+                customerEmail = "petya@email.com"
+            };
+            var customer2 = new Customer
+            {
+                cutomerName = "Vasya",
+                customerEmail = "vasya@email.com"
+            };
+            var customer3 = new Customer
+            {
+                cutomerName = "Kolya",
+                customerEmail = "kolya@email.com"
+            };
+
+            context.Customers.Add(customer1);
+            context.Customers.Add(customer2);
+            context.Customers.Add(customer3);
+
+            context.SaveChanges();
+        }
+        static void prepareSuppliers(Context context)
+        {
+            var supplier1 = new Supplier
+            {
+                supplierName = "Rozetka",
+                supplierPhone = "+38(098)-988-88-88",
+                supplierEmail = "supplier@rozetka.ua"
+            };
+            var supplier2 = new Supplier
+            {
+                supplierName = "Citrus",
+                supplierPhone = "+38(099)-999-09-09",
+                supplierEmail = "supplier@citrus.ua"
+            };
+
+            context.Suppliers.Add(supplier1);
+            context.Suppliers.Add(supplier2);
+
+            context.SaveChanges();
+        }
+        static void prepareProducts(Context context)
+        {
+            var suppliers = context.Suppliers.Take(2).ToArray();
```

Comment:
you can create suppliers here as well and just assign to the product field, then EF will save suppliers and products as well by one request without this trick

Reply:
Done in cd93496ab08d07c56a756b70c234cd198533c6a4